x-pack/solutions/security/plugins/security_solution/public/siem_migrations/rules/components/rule_details_flyout/tabs/translation/migration_rule_query.tsx

```diff
@@ -158,12 +161,73 @@ export const MigrationRuleQuery: React.FC<MigrationRuleQueryProps> = React.memo(
       );
     }, [editMode, form, onCancel, onSave]);
 
+    const esqlQueryBar = useMemo(() => {
+      if (!editMode) {
+        return null;
+      }
+      return (
+        <Form form={form} data-test-subj="ruleMigrationTranslationTab">
+          <EuiFlexGroup
+            direction="row"
+            gutterSize="none"
+            justifyContent="flexEnd"
+            alignItems="flexStart"
+          >
+            <EuiFlexItem grow={false}>
+              <EuiButtonEmpty iconType="cross" onClick={onCancel} size="xs">
+                {i18n.CANCEL}
+              </EuiButtonEmpty>
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <EuiButtonEmpty iconType="save" onClick={onSave} size="xs">
+                {i18n.SAVE}
+              </EuiButtonEmpty>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+          <CommonUseField
+            path="ruleName"
+            componentProps={{
+              idAria: 'ruleMigrationTranslationRuleName',
+              'data-test-subj': 'ruleMigrationTranslationRuleName',
+              euiFieldProps: {
+                fullWidth: true,
+              },
+            }}
+          />
+          <EuiFormRow id="queryEditor" data-test-subj="queryEsqlEditor" fullWidth>
+            <ESQLLangEditor
+              query={{ esql: query }}
+              onTextLangQueryChange={
+                (/* q: AggregateQuery */) => {
+                  // setQuery(q);
+                  // setParam('esqlQuery', q);
+                  // refreshTimeFields(q);
+                }
+              }
+              onTextLangQuerySubmit={async () => {}}
+              // detectedTimestamp={detectedTimestamp}
+              hideRunQueryText={true}
+              // isLoading={isLoading}
+              // editorIsInline
```

Comment:
`editorIsInline` allows to make initial height bigger, but exposes `Refresh` button